packages/project-access/src/project/cap.ts

```diff
@@ -0,0 +1,98 @@
+import { join } from 'path';
+import { FileName } from '../constants';
+import type { CapCustomPaths, CapProjectType, csn, Package } from '../types';
+import { fileExists, readJSON } from '../file';
+import { loadModuleFromProject } from './module-loader';
+
+/**
+ * Returns true if the project is a CAP Node.js project.
+ *
+ * @param packageJson - the parsed package.json object
+ * @returns - true if the project is a CAP Node.js project
+ */
+export function isCapNodeJsProject(packageJson: Package): boolean {
+    return !!(packageJson.cds || packageJson.dependencies?.['@sap/cds']);
+}
+
+/**
+ * Returns true if the project is a CAP Java project.
+ *
+ * @param projectRoot - the root path of the project
+ * @returns - true if the project is a CAP project
+ */
+export async function isCapJavaProject(projectRoot: string): Promise<boolean> {
+    const { srv } = await getCapCustomPaths(projectRoot);
+    return fileExists(join(projectRoot, srv, 'src', 'main', 'resources', 'application.yaml'));
+}
+
+/**
+ * Returns the CAP project type, undefined if it is not a CAP project.
+ *
+ * @param projectRoot - root of the project, where the package.json resides.
+ * @returns - CAPJava for Java based CAP projects; CAPNodejs for node.js based CAP projects; undefined if it is no CAP project
+ */
+export async function getCapProjectType(projectRoot: string): Promise<CapProjectType | undefined> {
+    if (await isCapJavaProject(projectRoot)) {
+        return 'CAPJava';
```

Comment:
Does it make sense to define a constant/enum for this one?

Reply:
@zdravko-georgiev we went away from enums to use basic types since I create less overhead when exporting. 
While the code now has a basic string, TS will throw errors in case you mistype.

Reply:
thanks for the info